### PR TITLE
[Merged by Bors] - feat: port Algebra.Lie.Submodule

### PR DESCRIPTION
**Pull request by int-y1**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Lie/Submodule.lean [hunk 1]
```diff
@@ -235,21 +236,25 @@ theorem lie_mem_left (I : LieIdeal R L) (x y : L) (h : x ∈ I) : ⁅x, y⁆ ∈
 #align lie_mem_left lie_mem_left
 
 /-- An ideal of a Lie algebra is a Lie subalgebra. -/
+-- Porting note: added `@[coe]` to fix `norm_cast` issues, but this causes bad pretty-printing.
```
Comment (jcommelin) [inline Mathlib/Algebra/Lie/Submodule.lean:239]: Could you please elaborate? What kind of `norm_cast` issues? What kind of bad pretty-printing? Is this a known issue?

**Review by jcommelin — COMMENTED** [review]
LGTM, aside from one question about a porting note.

File: Mathlib/Algebra/Lie/Submodule.lean (hunk 1 above)
Comment (int-y1) [reply Mathlib/Algebra/Lie/Submodule.lean:239]: @Ruben-VandeVelde

Comment (Ruben-VandeVelde) [reply Mathlib/Algebra/Lie/Submodule.lean:239]: `(I : LieSubalgebra R L)` becomes `(↑R L I)`

File: Mathlib/Algebra/Lie/Submodule.lean [hunk 2]
```diff
@@ -0,0 +1,1286 @@
...
+theorem lie_mem_right (I : LieIdeal R L) (x y : L) (h : y ∈ I) : ⁅x, y⁆ ∈ I :=
+  I.lie_mem h
+#align lie_mem_right lie_mem_right
+
+theorem lie_mem_left (I : LieIdeal R L) (x y : L) (h : x ∈ I) : ⁅x, y⁆ ∈ I := by
+  rw [← lie_skew, ← neg_lie]; apply lie_mem_right; assumption
+#align lie_mem_left lie_mem_left
+
+/-- An ideal of a Lie algebra is a Lie subalgebra. -/
+-- Porting note: added `@[coe]` to fix `norm_cast` issues, but this causes bad pretty-printing.
```
Comment (jcommelin) [inline Mathlib/Algebra/Lie/Submodule.lean:239]:
```suggestion
-- Porting note: added `@[coe]` to fix `norm_cast` issues, but this causes bad pretty-printing:
-- `(I : LieSubalgebra R L)` becomes `(↑R L I)`
```

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**